src/pe/export.rs

```diff
@@ -56,52 +56,79 @@ pub type ExportOrdinalTable = Vec<u16>;
+    pub export_name_pointer_table: Option<ExportNamePointerTable>,
+    pub export_ordinal_table: Option<ExportOrdinalTable>,
+    pub export_address_table: Option<ExportAddressTable>,
 }
 
 impl<'a> ExportData<'a> {
     pub fn parse(bytes: &'a [u8], dd: &data_directories::DataDirectory, sections: &[section_table::SectionTable]) -> error::Result<ExportData<'a>> {
         let export_rva = dd.virtual_address as usize;
         let size = dd.size as usize;
         debug!("export_rva {:#x} size {:#}", export_rva, size);
-        let export_offset = utils::find_offset_or(export_rva, sections, &format!("Cannot map export_rva ({:#x}) into offset", export_rva))?;
-        let export_directory_table = ExportDirectoryTable::parse(bytes, export_offset)?;
+        let export_offset = utils::find_offset_or(export_rva, sections, &format!("cannot map export_rva ({:#x}) into offset", export_rva))?;
+        let export_directory_table = ExportDirectoryTable::parse(bytes, export_offset)
+            .map_err(|_| error::Error::Malformed(format!("cannot parse export_directory_table (offset {:#x})", export_offset)))?;
         let number_of_name_pointers = export_directory_table.number_of_name_pointers as usize;
         let address_table_entries = export_directory_table.address_table_entries as usize;
-        //let ordinal_base = export_directory_table.ordinal_base as usize;
 
-        let name_pointer_table_offset = &mut utils::find_offset_or(export_directory_table.name_pointer_rva as usize, sections, &format!("Cannot map export_directory_table.name_pointer_rva ({:#x}) into offset", export_directory_table.name_pointer_rva))?;
-        let mut export_name_pointer_table: ExportNamePointerTable = Vec::with_capacity(number_of_name_pointers);
-        for _ in 0..number_of_name_pointers {
-            export_name_pointer_table.push(bytes.gread_with(name_pointer_table_offset, scroll::LE)?);
-        }
+        let export_name_pointer_table = utils::find_offset(export_directory_table.name_pointer_rva as usize, sections).map(|table_offset| {
+            let mut offset = table_offset;
+            let mut table: ExportNamePointerTable = Vec::with_capacity(number_of_name_pointers);
 
-        let export_ordinal_table_offset = &mut utils::find_offset_or(export_directory_table.ordinal_table_rva as usize, sections, &format!("Cannot map export_directory_table.ordinal_table_rva ({:#x}) into offset", export_directory_table.ordinal_table_rva))?;
-        let mut export_ordinal_table: ExportOrdinalTable = Vec::with_capacity(number_of_name_pointers);
-        for _ in 0..number_of_name_pointers {
-            export_ordinal_table.push(bytes.gread_with(export_ordinal_table_offset, scroll::LE)?);
-        }
+            for _ in 0..number_of_name_pointers {
+                if let Ok(name_rva) = bytes.gread_with(&mut offset, scroll::LE) {
+                    table.push(name_rva);
+                }
+                else {
+                    break;
```

Comment:
if i understand, we stop parsing all `name_pointers` here (and other `break` below for `export_address_table`) if a single one is bad, yes? is it possible to bump the offset by sizeof(name_rva), and continue?

This might not be a good idea for various reasons, and feel free not to implement this if its too much work, etc., since your PR already successfully parses my PE binary suite, so it would only be super-malformed binaries which isn't really a high priority.

Reply:
Ah, I think I get your point.

But the case where a `name_rva` cannot be read

```
for _ in 0..number_of_name_pointers {
  if let Ok(name_rva) = bytes.gread_with(&mut offset, scroll::LE) {
    table.push(name_rva);
    } else {
      break;
    }
}
```
there would be a serious error in the PE file for which data cannot be read from `bytes` at the given `offset` (e.g. the file is truncated), in such a case continuing to read would not help (personal opinion).